### PR TITLE
Jkarni/announce servant swagger

### PR DESCRIPTION
**Pull request by jkarni**
Not at all done, but opening a PR so it's easy to see where this is going and make comments.

cc @fizruk @dmjio

### CODE REVIEW

**Comment by jkarni** [comment]
@fizruk @dmjio you should both have push access now, if you want to add/make changes. I won't force-push for now (or push at all for the next few hours).

File: posts/2016-01-21-servant-swagger.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+There is `API Blueprint`, `RAML`, `Apiary`, and finally `swagger`. While these
+Web API Description languages don't do much in the way of helping you build web
+services in a type-safe way, they are generally very mature, and have some
+amazing tooling. For example, take a look at what `swagger-ui`, a client-side
+HTML, CSS, and JS bundle, does with your `swagger` API description:
+
+...
+
+This is a pretty nifty way of documenting your API! But it doesn't end there.
+If you have a `swagger` specification of you API, you can also take advantage
```
Comment (soenkehahn) [inline posts/2016-01-21-servant-swagger.md]: `your`

**Comment by soenkehahn** [comment]
Nice post.

**Comment by dmjio** [comment]
:+1: Great work

Might be worth mentioning the `Try it out!` button. (http://petstore.swagger.io/#!/pet/addPet).

> This is a pretty nifty way of documenting your API!
could be changed to
> This is a pretty nifty way of both documenting and exploring your API!

**Comment by jkarni** [comment]
Still intend to (a) finish up; (b) convert to literate haskell.

File: posts/2016-01-21-servant-swagger.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+library automatically. You don't even need to build the Java code - you can
+just use the "Generate Client" button in the beautiful
+[swagger editor](http://editor.swagger.io/#/).
+
+There are a wide array of other [tools](http://swagger.io/open-source-integrations/)
+that support `swagger`. Obviously, having access to them would be a great boon.
+The problem so far has been that writing and maintaining a `swagger`
+specification, that you are sure matches your service, isn't fun.
+
+# Swagger-servant
```
Comment (fizruk) [inline posts/2016-01-21-servant-swagger.md]: servant-swagger?

**Comment by jkarni** [comment]
"Try it out" doesn't work, but I can't quite say why (it's not CORS, it seems). Help appreciated!

**Comment by soenkehahn** [comment]
Here's the error message that I get (from `chromium`):
```
XMLHttpRequest cannot load https://hackage.haskell.org/user/SoenkeHahn. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://petstore.swagger.io' is therefore not allowed access.
```

**Comment by jkarni** [comment]
Hm, so it is CORS. Strange that I don't see that error message...

**Comment by dmjio** [comment]
JavaScript Ajax can't make cross domain calls I guess, unless you have a special header (Cross Origin resource sharing).

**Comment by dmjio** [comment]
Oh yea, try adding "Access-Allow-Origin:*" to the endpoint that serves swagger.json

**Pull request merged**